public/app/plugins/datasource/cloud-monitoring/components/AlignmentPeriods.tsx

```diff
@@ -31,6 +31,11 @@ export const AlignmentPeriods: FC<Props> = ({
     label: ap.text,
   }));
 
+  if (alignmentPeriod === 'stackdriver-auto') {
```

Comment:
Can we handle this on the backend instead?

Reply:
It is [handled](https://github.com/grafana/grafana/blob/3037bda44908e572411d7b1447f460307837a47e/pkg/tsdb/cloudmonitoring/cloudmonitoring.go#L309) in the backend. This is required here because the `stackdriver-auto` value is no more among the available alignment periods so its value [appears empty](https://github.com/grafana/grafana/blob/3037bda44908e572411d7b1447f460307837a47e/public/app/plugins/datasource/cloud-monitoring/components/AlignmentPeriods.tsx#L45) in the query editor.
An alternative would be to keep `stackdriver-auto` as hidden alignment period. I will try that.